src/tests/e2etest/E2ETest.cs

```diff
@@ -151,6 +158,35 @@ private T Deserialize<T>(TwinCollection twinCollection)
             }
         }
 
+        /// <summary>
+        /// Validates the local reported object and fails if not in sync with the given reported object.
+        /// </summary>
+        /// <param name="reported">The reported twin to check the local twin against</param>
+        /// <param name="componentName">The name of the component</param>
+        /// <param name="objectName">The object name (Optional)</param>
+        protected void ValidateLocalReported(Object reported, String componentName, String objectName = "")

```

Comment:
> protected void ValidateLocalReported(Object reported, String componentName, String objectName = "") [](http://example.com/codeflow?start=7&length=100)

This PR appears to only add coverage for the RC path. The DC path is not covered at all apparently. Is this inaccurate?

Two asks:

1. Need to validate that when something is changed via either DC or Desired Twin that both RC and Reported Twin are updated.

2. Need to validate that when something is changed first from DC and then Twin or vice versa that the RC and Reported Twin contain last respectively.